osquery/tables/networking/darwin/wifi.mm

```diff
@@ -132,10 +231,11 @@ QueryData genKnownWifiNetworks(QueryContext& context) {
     return {};
   }
 
-  boost::filesystem::path path = kAirPortPreferencesPath;
+  std::string p = isBigSurOrHigher() ? kAirPortPrefPathPostCatalina
+                                     : kAirPortPreferencesPath;
+  boost::filesystem::path path = p;
```

Comment:
Something feels a little weird here. Why assign p? It's just being set to a constant. Why call isBigSurOrHigher twice?

Why not:
```
if isBigSurOrHigher() {
  return parseBigSur(kAirPortPrefPathPostCatalina);
}
```

Reply:
Will this function apply for succeeding OS versions like Monterey?

Reply:
@lhmzhou Yes it will :)

Reply:
@directionless there is a bit of path validation, and didn't want to repeat the code. FWIW I am working on refactoring and cleaning up this part of code along with wifi_utils in a new PR, that will be cleaner.